packages/core/src/repositories/documentLogsRepository/index.ts

```diff
@@ -30,59 +26,35 @@ export class DocumentLogsRepository extends Repository<typeof tt, DocumentLog> {
       .as('documentLogsScope')
   }
 
-  async findByUuid(uuid: string | undefined) {
+  async findByUuid(uuid: string) {
     const result = await this.db
       .select()
       .from(this.scope)
-      .where(eq(this.scope.uuid, uuid ?? ''))
+      .where(eq(this.scope.uuid, uuid))
 
     if (!result.length) {
-      return Result.error(new NotFoundError('ProviderLog not found'))
+      return Result.error(
+        new NotFoundError(`DocumentLog not found with uuid ${uuid}`),
+      )
     }
 
     return Result.ok(result[0]!)
   }
 
-  async getDocumentLogsWithMetadata(
-    { documentUuid, draft }: { documentUuid: string; draft?: Commit },
-    tx = this.db,
-  ): Promise<TypedResult<DocumentLogWithMetadata[], NotFoundError>> {
-    const aggregatedFieldsSubQuery = tx
-      .select({
-        id: this.scope.id,
-        tokens: sum(providerLogs.tokens).mapWith(Number).as('tokens'),
-        cost_in_millicents: sum(providerLogs.cost_in_millicents)
-          .mapWith(Number)
-          .as('cost_in_millicents'),
-      })
-      .from(this.scope)
-      .innerJoin(
-        providerLogs,
-        eq(providerLogs.documentLogUuid, this.scope.uuid),
-      )
-      .groupBy(this.scope.id)
-      .as('aggregatedFieldsSubQuery')
-
-    const commitFilter = draft
-      ? or(isNotNull(commits.mergedAt), eq(commits.id, draft.id))
-      : isNotNull(commits.mergedAt)
-
-    const result = await tx
-      .select({
-        ...this.scope._.selectedFields,
-        commit: commits,
-        tokens: aggregatedFieldsSubQuery.tokens,
-        cost_in_millicents: aggregatedFieldsSubQuery.cost_in_millicents,
-      })
-      .from(this.scope)
-      .innerJoin(commits, eq(commits.id, this.scope.commitId))
-      .innerJoin(
-        aggregatedFieldsSubQuery,
-        eq(aggregatedFieldsSubQuery.id, this.scope.id),
-      )
-      .where(and(eq(this.scope.documentUuid, documentUuid), commitFilter))
-      .orderBy(desc(this.scope.createdAt))
-
-    return Result.ok(result)
+  async getDocumentLogsWithMetadata({
+    documentUuid,
+    draft,
+  }: {
+    documentUuid: string
+    draft?: Commit
+  }): Promise<TypedResult<DocumentLogWithMetadata[], Error>> {
+    return computeDocumentLogsWithMetadata(
```

Comment:
I just moved code around, this fetch is not returning the original repository scope which should be one of the constraints of repositories so I moved to its own service. Will probably remove this repository method at some point.